framework/src/samplers/SamplerInterface.C

```diff
@@ -27,12 +27,14 @@ SamplerInterface::SamplerInterface(const MooseObject * moose_object)
 {
 }
 
+template <>
 Sampler &
 SamplerInterface::getSampler(const std::string & name)
 {
-  return _smi_feproblem.getSampler(_smi_params.get<SamplerName>(name), _smi_tid);
+  return _smi_feproblem.getSampler(_smi_params.get<SamplerName>(name));
```

Comment:
If you supply a template argument then you get that type:

`MonteCarlo & mc = getSampler<MonteCarlo>(...)`

vs 

`Sampler & foo = getSampler(...)`

Reply:
Ah, so you don't want to have to do:
```
Sampler & foo = getSampler<>(...)
```
or 
```
Sampler & foo = getSampler<Sampler>(...)
```

Could this line change to just:
```
return getSampler<Sampler>(name)
```

Reply:
I want "getSampler(...)" to provide the base class to be consistent with other systems that return the base objects. But, we are needing to get specific Sampler types in many objects so rather than doing the conversion and check each time we need this, the template version is available.



> Could this line change to just:
> 
> ```
> return getSampler<Sampler>(name)
> ```

No, `getSampler<Sampler>(name)` is the function that this is defining, so that would be recursive.

Reply:
Ah, I see. Thank you for explaining.